message.go

```diff
@@ -1319,7 +1321,9 @@ func toMap(values []RedisMessage) map[string]RedisMessage {
 			continue
 		}
 		typ := values[i].typ
-		panic(fmt.Sprintf("redis message type %s as map key is not supported", typeNames[typ]))
+		return map[string]RedisMessage{
+			"error": {typ: typeSimpleErr, string: fmt.Sprintf("redis message type %s as map key is not supported", typeNames[typ])},
+		}
```

Comment:
We should return an error as the second return value here, instead of embedding it into the map.